### ISSUE TITLE
SCS dual variables to PSD constraints dont work

### ISSUE THREAD

**Issue by rileyjmurray** [cvxgrp/cvxpy#892]
With cvxpy 1.1 (alpha), and SCS 2.1.0.

If a user supplies a constraint ``con = X >> 0``, the optimal dual variable ``Y = con.dual_value`` should satisfy ``np.trace(X @ Y) <= eps`` for a small value of ``eps`` (e.g. ``1e-4`` or ``1e-8``). It should also be PSD.

Right now, neither of these properties hold. It seems SCS can return dual variables ``Y`` which are not symmetric, and where the symmetric part ``Y_sym = (Y + Y.T)/2.0`` is not PSD (the latter by a huge margin). 

I ran into this issue when refactoring tests in preparation for better dual variable support. Below is the problem data where I first encountered the error.
```
"""
Solve "Example 8.3" from Convex Optimization by Boyd & Vandenberghe.

Verify (1) optimal objective values, (2) that the dual variable to the PSD constraint
belongs to the correct cone (i.e. the dual variable is itself PSD), and (3) that
complementary slackness holds with the PSD primal variable and its dual variable.
"""
rho = cvx.Variable(shape=(4, 4), symmetric=True)
constraints = [0.6 <= rho[0, 1], rho[0, 1] <= 0.9,
               0.8 <= rho[0, 2], rho[0, 2] <= 0.9,
               0.5 <= rho[1, 3], rho[1, 3] <= 0.7,
                -0.8 <= rho[2, 3], rho[2, 3] <= -0.4,
               rho[0, 0] == 1, rho[1, 1] == 1,
               rho[2, 2] == 1, rho[3, 3] == 1,
               rho >> 0]
obj = cvx.Minimize(rho[0, 3])
prob = cvx.Problem(obj, constraints)
prob.solve(solver='SCS', verbose=True,eps=1e-8)
# ^ solved to high precision
mat = self.prob.constraints[-1].dual_value
sym_mat = (mat + mat.T) / 2.0
eigs = np.linalg.eigvalsh(sym_mat)
print(np.min(eigs))  # the smallest value is around -0.9.
```
If you see SCS acceleration errors, you can solve again with ``acceleration_lookback=0`` (possibly include ``warm_start=False``). You should still find that the recovered dual variable is not symmetric, and its symmetric part is very far from being PSD.

It's worth noting that from what I can tell, there are no tests in cvxpy (official repo, master branch) which check complementarity of SCS solutions for SDP.

**Comment by SteveDiamond** [comment]
Oops! Thanks for discovering this issue. Perhaps it broke in cvxpy 1.1. The PSD constraints for SCS are kind of tricky to interpret, and there used to be code to correct the dual variables. Maybe that disappeared.

**Comment by rileyjmurray** [comment]
I've fixed this on my fork; it'll be in my next PR. If you want, you can copy a gist from that PR to merge the fix separately.

**Comment by rileyjmurray** [comment]
Closed by #910
